File: src/0455.cpp
#include <vector>
#include <algorithm>

using namespace std;

class Solution {
public:
    int findContentChildren(vector<int>& g, vector<int>& s) {
        sort(g.begin(), g.end());
        sort(s.begin(), s.end());
        int pg = 0, ps = 0;
        while (ps < s.size() && pg < g.size()) {
            if (g[pg] <= s[ps]) {
                pg++;
            }
            ps++;
        }
        return pg;
    }
};

File: src/0455.py
class Solution:
    def findContentChildren(self, g: list[int], s: list[int]) -> int:
        g.sort()
        s.sort()
        pg = ps = 0
        while ps < len(s) and pg < len(g):
            if g[pg] <= s[ps]:
                pg += 1
            ps += 1
        return pg
